sdk/appcenter/src/test/java/com/microsoft/appcenter/channel/DefaultChannelTest.java

```diff
@@ -1190,4 +1197,33 @@ public void setNetworkRequest() {
         channel.setNetworkRequests(true);
         verify(mockIngestion, times(2)).sendAsync(anyString(), any(UUID.class), any(LogContainer.class), any(ServiceCallback.class));
     }
+
+    @Test
+    public void testConcurrentIterableDataStructureUpdatesSuccess() {
+        try {
+            Persistence mockPersistence = mock(Persistence.class);
+            AppCenterIngestion mockIngestion = mock(AppCenterIngestion.class);
+            Channel.GroupListener mockListener = mock(Channel.GroupListener.class);
+
+            //Get the concurrent hashmap in DefaultChannel that accepts modifications during iteration
+            DefaultChannel channel = new DefaultChannel(mock(Context.class), UUID.randomUUID().toString(), mockPersistence, mockIngestion, mAppCenterHandler);
+            Map<String, DefaultChannel.GroupState> myMap =  channel.getGroupStates();
+            channel.addGroup(TEST_GROUP, 50, BATCH_TIME_INTERVAL, MAX_PARALLEL_BATCHES, null, mockListener);
+            channel.addGroup(TEST_GROUP_TWO, 50, BATCH_TIME_INTERVAL, MAX_PARALLEL_BATCHES, null, mockListener);
+            channel.addGroup(TEST_GROUP_THREE, 50, BATCH_TIME_INTERVAL, MAX_PARALLEL_BATCHES, null, mockListener);
+
+            //Iterate over the map and modify twice
+            Iterator it1 = myMap.keySet().iterator();
+            while (it1.hasNext()) {
+                String key = it1.next().toString();
+                if(key.equals(TEST_GROUP_TWO)) {
```

Comment:
Please follow this https://www.oracle.com/technetwork/java/codeconventions-150003.pdf